engine/runtime-parser/src/main/scala/org/enso/compiler/core/ir/expression/errors/Syntax.scala

```diff
@@ -147,6 +147,11 @@ object Syntax {
       s"Syntax is not supported yet: $syntaxName"
   }
 
+  case object InconsistentConstructorVisibility extends Reason {
+    override def explanation: String =
+      "All constructors of a type must be either public or private"
```

Comment:
tbh I'm not sure if that is clear enough: I could understand this sentence as "∀c∈constructors(T) ∃p∈{public,private} c is p", whereas what we really want is "∃p∈{public,private} ∀c∈constructors(T) c is p".

What if we do
```suggestion
      "Constructors of a type must either all be private or all be public"
```
OR maybe even better:
```suggestion
      "Private and public constructors cannot be mixed within a single type"
```